docs/scenarios_authoring/DataSourcesAndSinks.md

```diff
@@ -184,17 +184,25 @@ Holds event in the node until
 The `key` parameter will be removed in the future release of Nussknacker, for the time being, configure it to `#inputMeta.key`.
 
 &nbsp;
-### Sample generator
+### Event generator
 
 **(Flink engine only)**
 
-![sample_generator_window](img/sample-generator.png)
+![event_generator_window](img/event-generator.png)
+
+This source generates events at a user-defined schedule.
+
+There are two primary use cases for this component:
+
+- Generating a stream of data (typically for testing): The `#input` variable, available downstream, will be populated
+  with the data generated by this component. Use the `#RANDOM` helper to produce random values.
+- Periodically generating a trigger event: This enables your scenario to run periodically, unlocking additional use
+  cases beyond the traditional stream processing.
 
-This source provides functionality of sending a number of given events in a periodic way. It's mainly used for testing.
+Parameters:
 
-This source has the following parameters:
-- period - specifies how often events will be sent
-- count - specifies number of event that will be sent every `period`
-- value - specifies data that event will hold
+- schedule - specifies how often events will be sent
+- count - specifies number of events that will be generated at every run
+- value - specifies data that event will hold. Expressions will be evaluated for every event in the `count`
```

Comment:
> in the `count`
Does this mean that an event is created once for each batch? Perhaps it would be better to remove this fragment?

Reply:
I think this may be not clear enough. What this was meant to express is that each event is evaluated seperately (so for count = 10 it will evaluate the expression 10 times, 1 per 1 count). Previously it worked differently. I'll talk with this paragraph's author.

Reply:
Okay, maybe `specifies data that event will hold. Expressions will be evaluated for every generated/emitted event` is good enough?

Reply:
Right, its ok. Changed it like you proposed just with `Expression` not `Expressions` since its just one expression thats being evaluated